src/main/src/test/java/org/vfny/geoserver/global/IndexComparatorsTest.java

```diff
@@ -0,0 +1,88 @@
+/* (c) 2014 Open Source Geospatial Foundation - all rights reserved
+ * (c) 2001 - 2013 OpenPlans
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.vfny.geoserver.global;
+
+import java.io.IOException;
+import java.util.Set;
+import java.util.TreeSet;
+import org.geoserver.catalog.FeatureTypeInfo;
+import org.geoserver.catalog.ResourceInfo;
+import org.geoserver.catalog.impl.CoverageInfoImpl;
+import org.geoserver.catalog.impl.FeatureTypeInfoImpl;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Tests covering the former functionality of GeoServerDataDirectory.
+ *
+ * <p>Much of this functionality depends on the availability of GeoServerResourceLoader in the
+ * application context as the bean "resourceLoader".
+ *
+ * @author Antonio Cerciello - Geocat
+ */
+public class IndexComparatorsTest {
+
+    @Test
+    public void TestFeatureTypeSort() throws IOException {
+
+        Set<FeatureTypeInfo> unordered =
```

Comment:
Confusing naming, a "TreeSet" is ordered by definition. At a first glance I thought you were adding elements in a list and was wondering where the call to Collections.sort was :-D

Reply:
I feel very guilty about this... I don't know what I was thinking! :smile: